packages/database/test/exp/integration.test.ts

```diff
@@ -97,6 +105,39 @@ describe('Database@exp Tests', () => {
     expect(snap2).to.equal('b');
   });
 
+  // Tests to make sure onValue's data does not get mutated after calling get
+  it('calls onValue only once after get request with a non-default query', async () => {
+    const db = getDatabase(defaultApp);
+    const testRef = ref(db, 'foo');
+    const initial = [{ name: 'child1' }, { name: 'child2' }];
+
+    let count = 0;
+    await set(testRef, initial);
+    onValue(testRef, snapshot => {
```

Comment:
I think these tests could be refactored into table-tests over a spec like this:

```
{
  "initialData": [{ name: 'child1' }, { name: 'child2' }],
  "initialLocation": <some path>,
  "listenerFunc": either onChild{Added, Changed, Removed} or onValue,
  "listenerQuery": <the query to listen to>,
  "expectedListenResults": [<array of snaps>] # fail if exactly these aren't received"
  "getQuery": <the query to get>,
   "expectedGetResults": <snap>, # fail if exactly this one isn't received.
  
}
```

Then we could easily add similar tests to the ones added here and check for various combinations of listens and gets (listen above get, below get, at get, etc.).

However, as I'm wrapping up this comment I suspect this refactor deserves it's own PR so feel free to ignore.